azurerm/internal/services/policy/tests/resource_arm_policy_remediation_test.go

```diff
@@ -0,0 +1,733 @@
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMPolicyRemediationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMPolicyRemediation_atResource(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPolicyRemediationExists(data.ResourceName),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "scope"),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "policy_assignment_id"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMPolicyRemediation_updateLocation(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_policy_remediation", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMPolicyRemediationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMPolicyRemediation_atResourceGroup(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPolicyRemediationExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "location_filters.#", "0"),
+				),
+			},
+			{
+				Config: testAccAzureRMPolicyRemediation_updateLocation(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPolicyRemediationExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "location_filters.#", "1"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMPolicyRemediation_requiresImport(t *testing.T) {
+	if !features.ShouldResourcesBeImported() {
+		t.Skip("Skipping since resources aren't required to be imported")
+		return
+	}
+
+	data := acceptance.BuildTestData(t, "azurerm_policy_remediation", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMPolicyRemediationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMPolicyRemediation_atResourceGroup(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPolicyRemediationExists(data.ResourceName),
+				),
+			},
+			data.RequiresImportErrorStep(testAccAzureRMPolicyRemediation_requiresImport),
+		},
+	})
+}
+
+func testCheckAzureRMPolicyRemediationExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Policy Insights Remediation not found: %s", resourceName)
+		}
+
+		id, err := parse.PolicyRemediationID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		client := acceptance.AzureProvider.Meta().(*clients.Client).Policy.RemediationsClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
```

Comment:
could we move these to the top of the anon function?

Reply:
Sure!